File: two pointers/Container With Most Water.py
def maxArea(height: list[int]) -> int:
    l, r = 0, len(height) - 1
    ans = 0

    while l < r:
        lh, rh = height[l], height[r]
        area = min(lh, rh) * (r - l)
        ans = max(ans, area)

        if lh < rh:
            l += 1
        else:
            r -= 1


    return ans

print(maxArea([1,8,6,2,5,4,8,3,7]))
print(maxArea([1,2,1]))

File: two pointers/Container With Most Water.cs
public class SolutionMaxArea
{
    public int MaxArea(int[] height)
    {
        int ans = 0;
        int l = 0;
        int r = height.Length - 1;

        while (l < r)
        {
            int lH = height[l];
            int rH = height[r];
            int water = (r - l) * Math.Min(rH, lH);
            if (lH < rH)
                ++l;
            else
                --r;

            ans = Math.Max(ans, water);
        }
        return ans;
    }
}
